app/views/layouts/_header.html.erb

```diff
@@ -104,7 +104,7 @@
           <%= link_to t("layouts.logout"), logout_path(:referer => request.fullpath), :method => "post", :class => "geolink dropdown-item" %>
         </div>
       </div>
-    <% else %>
+    <% elsif controller_name != "users" and controller_name != "sessions" %>
```

Comment:
This means that if you go to users#show (e.g. visit www.osm.org/user/foo while not logged in) then there is no button to log in or sign up.

If this problem only affects the login and signup pages it needs to be more targetted, either by specific controller methods or moving the signup code into a different controller (e.g. RegistrationsController, to align with Devise). 

Alternatively, perhaps there is a better design solution, since removing these buttons on the full-width layout looks weird and slightly inconsistent to me.

Reply:
Good catch, thanks.

I will modify the condition to target only `new` action in sessions_controller and users_controller. Hopefully that covers it. Otherwise, I would revert the modifications to _hedaer.html.erb,  and address the "noisy" Login/Signup buttons from the header in a separate PR.

Reply:
I'd show/hide the buttons only in small layout:
https://github.com/openstreetmap/openstreetmap-website/commit/09877a78e901698e10e8172ed868395653767ae6

Reply:
@AntonKhorev Interesting. It looks good to me, if we desire to keep the header buttons and tabs at the same time.

@gravitystorm
> Alternatively, perhaps there is a better design solution, since removing these buttons on the full-width layout looks weird and slightly inconsistent to me.

There is a small inconsistency if we keep Login/Signup on the pages along with the tabs under the globe: The login buttons have different referrer:
- login button in adds the referrer to the current page
- login button in the tab keeps the referrer as originally set by initial "login" button that invoked it.